packages/reactive-dict/reactive-dict.js

```diff
@@ -164,6 +164,23 @@ _.extend(ReactiveDict.prototype, {
     
   },
 
+  delete: function(key) {
+    var self      = this;
+    var oldKeys   = self.keys;
+    var didRemove = false;
+
+    if (_.has(oldKeys, key)) {
+      changed(self.keyDeps[key]);
+      changed(self.keyValueDeps[key][oldKeys[key]]);
+      changed(self.keyValueDeps[key]['undefined']);
+      delete self.keys[key]; // clean up
```

Comment:
A Computation can listen on any key, whether or not it exists in the dict, and we still track it.  So removing a key doesn't mean we should go and remove the Dependency objects.  However, it seems like we should _eventually_ clean up the keys.  We currently clean up the values in `keyValueDeps` but not the keys (the second level of the map but not the first), and we never clean up `keyDeps`.  I'm not sure where to put the clean-up code.